algorithms/src/snark/plookup/table.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkvm_fields::Field;
+
+pub struct LookupTable<F: Field>(Vec<Vec<F>>);
+
+impl<F: Field> LookupTable<F> {
+    pub fn new(rows: usize, cols: usize) -> Self {
+        Self(vec![vec![F::zero(); cols]; rows])
+    }
+
+    pub fn push(&mut self, row: usize, col: usize, element: F) {
```

Comment:
this name might suggest that the table grows in the process; I'd suggest something like `fill` instead